lib/rucio/tests/test_abacus_collection_replica.py

```diff
@@ -145,14 +148,72 @@ def test_abacus_collection_replica(self):
         assert str(dataset_replica['state']) == 'UNAVAILABLE'
 
         # Delete all files -> collection replica should be deleted
-        from rucio.daemons.reaper.reaper import REGION
-        REGION.invalidate()
+        # Old behaviour (doesn't delete the DID)
         cleaner.run(once=True)
+        reaper.REGION.invalidate()
         if self.vo:
             reaper.run(once=True, include_rses='vo=%s&(%s)' % (self.vo['vo'], self.rse), greedy=True)
         else:
             reaper.run(once=True, include_rses=self.rse, greedy=True)
         self.rule_client.add_replication_rule([{'scope': self.scope, 'name': self.dataset}], 1, self.rse, lifetime=-1)
         collection_replica.run(once=True)
         dataset_replica = [replica for replica in self.replica_client.list_dataset_replicas(self.scope, self.dataset)]
-        assert len(dataset_replica) == 0
+        assert dataset_replica[0]['length'] == 0
+        assert dataset_replica[0]['available_length'] == 0
+
+
+@pytest.mark.parametrize("core_config_mock", [{"table_content": [
+    ('reaper', 'remove_open_did', True)
+]}], indirect=True)
+@pytest.mark.parametrize("caches_mock", [{"caches_to_mock": [
+    'rucio.core.config', 'rucio.core.replica'
+]}], indirect=True)
+def test_abacus_collection_replica_new(vo, rse_factory, rucio_client, did_factory, core_config_mock, caches_mock):
```

Comment:
I wouldn't use the "_new" suffix. Probably better to name the test by its function. For example : _with_open_did_cleanup ; or something like that.

Reply:
same in all other places where you refer to "old behavior" or "new"